src/main/java/ch/njol/skript/effects/EffFireworkLaunch.java

```diff
@@ -0,0 +1,84 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.effects;
+
+import org.bukkit.FireworkEffect;
+import org.bukkit.Location;
+import org.bukkit.entity.Firework;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.meta.FireworkMeta;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+@Name("Launch firework")
+@Description("Leash firework effects at the given location(s).")
+@Examples("leash the player to the target entity")
+@Since("INSERT VERSION")
+public class EffFireworkLaunch extends Effect {
+	
+	static {
+		Skript.registerEffect(EffFireworkLaunch.class, "(launch|deploy) [[a] firework [with effect[s]]] %fireworkeffects% at %locations% [(with duration|timed) %number%]");
```

Comment:
This should use **`[%directions% %locations%]`**, example in [EffSpawn](https://github.com/SkriptLang/Skript/blob/master/src/main/java/ch/njol/skript/effects/EffSpawn.java#L70).
Optional location and the **at** comes from the direction type.
Also `timed/duration %number%` doesn't make sense. Making it **`with [flight] power %-number%`** would be better if you note that 1 power = 0.5 seconds. Also you forgot to make the expression nullable.